neo/rawio/intanbinaryrawio.py

```diff
@@ -0,0 +1,438 @@
+
+from .baserawio import (
+    BaseRawIO,
+    _signal_channel_dtype,
+    _signal_stream_dtype,
+    _spike_channel_dtype,
+    _event_channel_dtype,
+    _common_sig_characteristics,
+)
+
+from .intanrawio import (
+    read_variable_header,
+    rhd_global_header_base,
+    rhd_global_header_part1,
+    rhd_global_header_v11,
+    rhd_global_header_v13,
+    rhd_global_header_v20,
+    rhd_global_header_final,
+    rhd_signal_group_header,
+    rhd_signal_channel_header,
+    stream_type_to_name,
+)
+
+
+class IntanBinaryRawIO(BaseRawIO):
+    """
+    Class for processing Intan Data when saved in binary format. Requires an
+    `info.rhd` as well one file per signal stream or one file per channel
+
+    Parameters
+    ----------
+    dirname: str, Path
+        The root directory containing the info.rhd file
+
+    """
+
+    extensions = ["rhd", "dat"]
+    rawmode = "one-dir"
+
+    def __init__(self, dirname=""):
+        BaseRawIO.__init__(self)
+        self.dirname = dirname
+
+    def _source_name(self):
+        return self.dirname
+
+    def _parse_header(self):
+
+        dir_path = Path(self.dirname)
+        assert dir_path.is_dir(), (
+            "IntanBinaryRawIO requires the root " "directory containing info.rhd"
+        )
+
+        header_file = dir_path / "info.rhd"
+
+        for file in possible_raw_files:
+            if (dir_path / file).is_file():
+                one_file_per_signal = True
+                break
+            else:
+                one_file_per_signal = False
+
+        self.one_file_per_signal = one_file_per_signal
+
+        if one_file_per_signal:
+            raw_file_dict = create_one_file_per_signal_dict(dir_path)
+        else:
+            raw_file_dict = create_one_file_per_channel_dict(dir_path)
+
+        (
+            self._global_info,
+            self._ordered_channels,
+            data_dtype,
+            self._block_size,
+        ) = read_rhd(header_file)
+
+        self._raw_data = {}
+        for stream_index, sub_datatype in data_dtype.items():
+            if one_file_per_signal:
+                self._raw_data[stream_index] = np.memmap(
+                    raw_file_dict[stream_index], dtype=sub_datatype, mode="r"
+                )
+            else:
+                self._raw_data[stream_index] = []
+                for channel_index, datatype in enumerate(sub_datatype):
+                    self._raw_data[stream_index].append(
+                        np.memmap(
+                            raw_file_dict[stream_index][channel_index],
+                            dtype=[datatype],
+                            mode="r",
+                        )
+                    )
+
+        # check timestamp continuity
+        if one_file_per_signal:
+            timestamp = self._raw_data[6]["timestamp"].flatten()
+        else:
+            timestamp = self._raw_data[6][0]["timestamp"].flatten()
+        assert np.all(np.diff(timestamp) == 1), "timestamp have gaps"
```

Comment:
Why are we stoping the user from working here?

Is there anything the user can do in these cases? If so, maybe we can include in the assertion message.

Reply:
That was a holdover from the original RawIO. There are two reasons this can occur:

File corruption (start-stops, run out of storage). In this case I don't know if we can really salvage the files?
Attempting to merge files that shouldn't be merged (ie spread out in time). This is something SI does all the time for recordings, so I don't think this is wrong to do. 

Now that I think about this, this may be a better check when all data is in the same `.rhd` file rather than these headerless binary files. I'll think about this.

Reply:
OK. Sorry, I did not check the original. I should have done that. Let's get this working and then we discuss this type of issues after this PR as they will require modification -if they do require modification- of the two classess.  Right now, better to keep them in synch.